zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -522,68 +517,122 @@ def get_local_osd_id(self, unit):
         local = list(json.loads(ret['Stdout']))[-1]
         return local if local.startswith('osd.') else 'osd.' + local
 
-    def get_num_osds(self, osd):
+    def get_num_osds(self, osd, is_up_only=False):
         """Compute the number of active OSD's."""
         result = zaza_model.run_on_unit(osd, 'ceph osd stat --format=json')
         result = json.loads(result['Stdout'])
-        return int(result['num_osds'])
+        if is_up_only:
+            return int(result['num_up_osds'])
+        else:
+            return int(result['num_osds'])
+
+    def get_osd_devices_on_unit(self, unit_name):
+        """Get osd devices on a unit."""
+        osd_devices = json.loads(
+            zaza_model.run_on_unit(
+                unit_name, 'ceph-volume lvm list --format=json'
+            ).get('Stdout', '')
+        )
+
+        return osd_devices
+
+    def remove_disk_from_osd_unit(self, unit, osd_id, is_purge: bool = False):
+        """Remove provided osd_id from unit."""
+        action_obj = zaza_model.run_action(
+            unit_name=unit,
+            action_name='remove-disk',
+            action_params={
+                'osd-ids': osd_id, 'timeout': 5,
+                'format': 'json', 'purge': is_purge
```

Comment:
From a code-reading perspective, this might be nicer as 4 lines.  i.e. then it's obvious that there are 4 items in the dictionary, etc.

Reply:
ack